File: WindowsFormsApp2/MapService.VB/MapService.asmx.vb
'
' Copyright (c) 2014-2020, MindFusion LLC - Bulgaria
'

Imports System.Web.Services
Imports System.Web.Services.Protocols
Imports System.ComponentModel
Imports MindFusion.Mapping.WebForms

' To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line.
<System.Web.Script.Services.ScriptService()> _
<System.Web.Services.WebService(Namespace:="http://tempuri.org/")> _
<System.Web.Services.WebServiceBinding(ConformsTo:=WsiProfiles.BasicProfile1_1)> _
<ToolboxItem(False)> _
Public Class MapService
    Inherits System.Web.Services.WebService

    <WebMethod()> _
    Public Function GenerateMap(ByVal json As String) As String
        Return MapView.GenerateMap(json)
    End Function

End Class

File: WindowsFormsApp2/MapService.CS/MapService.asmx.cs
//
// Copyright (c) 2014-2020, MindFusion LLC - Bulgaria
//

using System.Web.Services;
using MindFusion.Mapping.WebForms;

namespace MapService
{
    /// <summary>
    /// Handles generation of map images.
    /// </summary>
    [WebService(Namespace = "http://tempuri.org/")]
    [WebServiceBinding(ConformsTo = WsiProfiles.BasicProfile1_1)]
    [System.ComponentModel.ToolboxItem(false)]
    // To allow this Web Service to be called from script, using ASP.NET AJAX, uncomment the following line. 
    [System.Web.Script.Services.ScriptService]
    public class MapService : System.Web.Services.WebService
    {
        [WebMethod]
        public string GenerateMap(string json)
        {
            return MapView.GenerateMap(json);
        }
    }
}
